### PR TITLE
Updated the code to navigate to the product page after scanning(#1z59ah8)

### LINKED ISSUE #113: On successful scan user should be navigated to count/product page
## What is the motivation for adding/enhancing this feature?

As the scanner returns a single product only, we should move it to the details page, improving the flow and user experience.

## What are the acceptance criteria?

- [ ] When scanning a product, the user should be directly navigated to the details page

## Can you complete this feature request by yourself?
- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #113

### Short Description and Why It's Useful

For a better UX, the search bar and the searched product list will be cleared after the successful scan of an SKU.

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)
- [ ] I have changed the target to hacktoberfest branch

### CODE REVIEW

File: changelogs/unreleased/-1z59ah8.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
```
Comment (ymaheshwari1) [inline changelogs/unreleased/-1z59ah8.yml:1]: Remove changelog entry file

File: src/views/Search.vue [hunk 2]
```diff
@@ -97,11 +98,14 @@ export default defineComponent({
...
-        .then((result) => {
+        .then(async (result) => {
           //result : value of the scanned barcode/QRcode
-          if(result.role){
+          if(result.role && result.role !== 'backdrop'){
             this.queryString = result.role
-            this.getProducts(process.env.VUE_APP_VIEW_SIZE, 0);
+            await this.getProducts(process.env.VUE_APP_VIEW_SIZE, 0);
+            this.router.push({ path: `/count/${this.products[0].sku}` });
+            this.queryString = '';
```
Comment (ymaheshwari1) [inline src/views/Search.vue:107]: We can instead just don't assign a value to `this.queryString` on line 104.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Search.vue [hunk 3]
```diff
@@ -97,11 +98,18 @@ export default defineComponent({
...
           //result : value of the scanned barcode/QRcode
-          if(result.role){
+          if(result.role && result.role !== 'backdrop'){
             this.queryString = result.role
-            this.getProducts(process.env.VUE_APP_VIEW_SIZE, 0);
+            await this.getProducts(process.env.VUE_APP_VIEW_SIZE, 0);
+            this.router.push({ path: `/count/${this.products[0].sku}` });
+            // We are using the value in getProducts function above, and hence
+            // are emptying it after usage only.
+            this.queryString = '';
```
Comment (adityasharma7) [inline src/views/Search.vue:109]: I think this needs to do differently, we tried to use the existing method which is quite good, but as the flow works differently we should have a new implementation for the scanning and query search.

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Comment by ymaheshwari1** [comment]
Dependent on #56

**Comment by disha1202** [comment]
Not needed, closing

**Pull request closed without merging**